src/librbml/lib.rs

```diff
@@ -167,6 +275,15 @@ pub mod reader {
         pub next: uint
     }
 
+    pub fn tag_at(data: &[u8], start: uint) -> DecodeResult<Res> {
+        let v = data[start] as uint;
+        if v < 0xf1 { // 0xf0 is intentionally left out
```

Comment:
The decoder does not allow 0xf0 in the decoder, elimintaing the source of non-canonical encoding (tag 0x12 can be, for example, represented either by `12` or `f0 12`). Therefore it decodes `f0` as simply tag 0xf0, which should not be used, and will fail on the tag matching. This might be too far-reached thought and maybe I should just implement an accurate error reporting though.